deprecation warning on Android: 'slCreateEngine' is deprecated: first deprecated in Android 30

```
brenton@Brentons-M3-MacBook-Pro % ./gradlew :app:buildCMakeDebug

> Task :app:buildCMakeDebug[arm64-v8a]-2
C/C++: ninja: Entering directory `/Users/brenton/development/github/spacehockey/app/.cxx/Debug/4f1a6f4b/arm64-v8a'
C/C++: /Users/brenton/Library/Android/sdk/ndk/29.0.14206865/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang --target=aarch64-none-linux-android34 --sysroot=/Users/brenton/Library/Android/sdk/ndk/29.0.14206865/toolchains/llvm/prebuilt/darwin-x86_64/sysroot -DDEBUG -DGL_GLEXT_PROTOTYPES -DSDL_BUILD_MAJOR_VERSION=3 -DSDL_BUILD_MICRO_VERSION=0 -DSDL_BUILD_MINOR_VERSION=5 -DSDL_STATIC_LIB -DUSING_GENERATED_CONFIG_H -I/Users/brenton/development/github/spacehockey/app/.cxx/Debug/4f1a6f4b/arm64-v8a/_deps/sdl3-build/include-config-debug/build_config -I/Users/brenton/development/github/spacehockey/app/.cxx/Debug/4f1a6f4b/arm64-v8a/_deps/sdl3-build/include-revision -I/Users/brenton/development/github/SDL/include -I/Users/brenton/development/github/SDL/src -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -fsanitize=hwaddress -fno-omit-frame-pointer -D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security  -fno-limit-debug-info  -fPIC -Wall -Wundef -Wfloat-conversion -fno-strict-aliasing -Werror=documentation -Wdocumentation -Werror=documentation-unknown-command -Wdocumentation-unknown-command -fcomment-block-commands=threadsafety -Wshadow -Wno-unused-local-typedefs -Wimplicit-fallthrough -Werror -fcolor-diagnostics -idirafter/Users/brenton/development/github/SDL/src/video/khronos -MD -MT _deps/sdl3-build/CMakeFiles/SDL3-static.dir/src/audio/openslES/SDL_openslES.c.o -MF _deps/sdl3-build/CMakeFiles/SDL3-static.dir/src/audio/openslES/SDL_openslES.c.o.d -o _deps/sdl3-build/CMakeFiles/SDL3-static.dir/src/audio/openslES/SDL_openslES.c.o -c /Users/brenton/development/github/SDL/src/audio/openslES/SDL_openslES.c
C/C++: /Users/brenton/development/github/SDL/src/audio/openslES/SDL_openslES.c:147:14: error: 'slCreateEngine' is deprecated: first deprecated in Android 30 [-Werror,-Wdeprecated-declarations]
C/C++:   147 |     result = slCreateEngine(&engineObject, 0, NULL, 0, NULL, NULL);
C/C++:       |              ^
C/C++: /Users/brenton/Library/Android/sdk/ndk/29.0.14206865/toolchains/llvm/prebuilt/darwin-x86_64/sysroot/usr/include/SLES/OpenSLES.h:2752:28: note: 'slCreateEngine' has been explicitly marked deprecated here
C/C++:  2752 | SL_API SLresult SLAPIENTRY slCreateEngine(
C/C++:       |                            ^
C/C++: 1 error generated.

> Task :app:buildCMakeDebug[arm64-v8a]-2 FAILED

...

brenton@Brentons-M3-MacBook-Pro % 
```

--- a/src/audio/openslES/SDL_openslES.c
+++ b/src/audio/openslES/SDL_openslES.c
@@ -33,6 +33,11 @@
 #include <SLES/OpenSLES_Android.h>
 #include <android/log.h>
 
+// OpenSL ES is deprecated, but we still support it for now.
+#ifdef HAVE_GCC_DIAGNOSTIC_PRAGMA
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#endif // HAVE_GCC_DIAGNOSTIC_PRAGMA
 
 #define NUM_BUFFERS 2 // -- Don't lower this!
 
@@ -804,5 +809,9 @@ void OPENSLES_PauseDevices(void)
         }
     }
 }
+
+#ifdef HAVE_GCC_DIAGNOSTIC_PRAGMA
+#pragma GCC diagnostic pop
+#endif // HAVE_GCC_DIAGNOSTIC_PRAGMA
 
 #endif // SDL_AUDIO_DRIVER_OPENSLES

